app/controllers/check_records/check_records_controller.rb

```diff
@@ -16,5 +19,13 @@ class CheckRecordsController < ApplicationController
     def current_namespace
       "check-a-teachers-record"
     end
+
+    def enforce_terms_and_conditions_acceptance!
+      if current_dsi_user &&
+        (FeatureFlags::FeatureFlag.active?("terms_and_conditions") &&
+          current_dsi_user.terms_and_conditions_version_accepted != CURRENT_TERMS_AND_CONDITIONS_VERSION)
+          redirect_to check_records_terms_and_conditions_path
```

Comment:
There's still no logical check here based on the datetime they accepted the agreement.